### PR TITLE
[AKS] Add cluster-autoscaler profile

### PR DESCRIPTION
**Pull request by marwanad**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yungezz** [comment]
For review only, service not ready for merge.

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 1]
```diff
@@ -249,3 +250,28 @@ def validate_vnet_subnet_id(namespace):
...
+    """
+    _extract_cluster_autoscaler_params(namespace)
+    if namespace.cluster_autoscaler_profile is not None:
+        for key in namespace.cluster_autoscaler_profile.keys():
+            _validate_cluster_autoscaler_key(key)
+
+
+def _validate_cluster_autoscaler_key(key):
+    if not key:
+        raise CLIError('Empty key specified for cluster-autoscaler parameter')
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py]: paramter name is `cluster-autoscaler-profile`, please change it in the error message to keep consistent.

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 2]
```diff
@@ -370,6 +373,12 @@
   - name: --max-count
     type: int
     short-summary: Maximum nodes count used for autoscaler, when "--enable-cluster-autoscaler" specified. Please specify the value in the range of [1, 100]
+  - name: --clear-cluster-autoscaler-profile
+    type: bool
+    short-summary: Clears the cluster-autoscaler profile.
+  - name: --cluster-autoscaler-profile
+    type: list
+    short-summary: Space-separated list of key=value pairs for configuring cluster autoscaler.
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py]: Maybe you use empty string to represent clear operation. Then we can delete --clear-cluster-autoscaler-profile and only reserve --cluster-autoscaler-profile. For example, `--cluster-autoscaler-profile ""`

Comment (marwanad) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py]: If we go with this, with that example a user does:
```
az aks update {{some stuff}}
az aks update --cluster-autoscaler-profile ""
```
I could be missing something here but how do we differentiate between both intentions in the validator? To the validator in both cases, `namespace.cluster_autoscaler_profile` is falsy but the intention in the first command is to not clear the profile while the intention of the second one is to clear it.

**Comment by marwanad** [comment]
made some changes, pending the new API vendor - CI won't pass.

File: src/azure-cli/azure/cli/command_modules/acs/_help.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py]: This is the only request of mine. After this is revolved, I'll approve.

**Comment by marwanad** [comment]
will reopen after rebase.

**Pull request closed without merging**